go/host/db/batches.go

```diff
@@ -2,106 +2,121 @@ package db
 
 import (
 	"bytes"
+	"errors"
+
+	"github.com/obscuronet/go-obscuro/go/common/errutil"
 
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/obscuronet/go-obscuro/go/common"
-	"github.com/obscuronet/go-obscuro/go/common/log"
 )
 
 // DB methods relating to batches.
 
-// GetHeadBatchHeader returns the header of the node's current head batch, or (nil, false) if no such header is found.
-func (db *DB) GetHeadBatchHeader() (*common.Header, bool) {
-	headBatchHash, found := db.readHeadBatchHash()
-	if !found {
-		return nil, false
+// GetHeadBatchHeader returns the header of the node's current head batch.
+func (db *DB) GetHeadBatchHeader() (*common.Header, error) {
+	headBatchHash, err := db.readHeadBatchHash()
+	if err != nil {
+		return nil, err
 	}
 	return db.readBatchHeader(*headBatchHash)
 }
 
 // AddBatchHeader adds a batch's header to the known headers
-func (db *DB) AddBatchHeader(header *common.Header, txHashes []common.TxHash) {
+func (db *DB) AddBatchHeader(header *common.Header, txHashes []common.TxHash) error {
 	b := db.kvStore.NewBatch()
-	db.writeBatchHeader(header)
+	err := db.writeBatchHeader(header)
+	if err != nil {
+		return err
```

Comment:
Matter of preference, but I typically wrap the errors with a reason in methods that may return more than one error.

Reply:
Done